python/pyarrow/tests/test_flight.py

```diff
@@ -263,13 +316,13 @@ def flight_server(server_base, *args, **kwargs):
     def _server_thread():
         server_instance.run(location, auth_handler=auth_handler)
 
-    thread = threading.Thread(target=_server_thread, daemon=True)
-    thread.start()
+    server_process = multiprocessing.Process(target=_server_thread)
+    server_process.start()
 
     yield location
```

Comment:
Should have a `try...finally` around this so that the process is always terminated.